src/main/java/com/artipie/metrics/memory/InMemoryMetrics.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 artipie.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.metrics.memory;
+
+import com.artipie.metrics.Metrics;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+/**
+ * {@link Metrics} implementation storing data in memory.
+ *
+ * @since 0.9
+ * @todo #231:30min Support gauges in InMemoryMetrics.
+ *  `InMemoryMetrics.gauge()` method implementation should get or create an `InMemoryGauge` by name
+ *  and store it. `InMemoryMetrics.counters()` method should be added
+ *  to create snapshot of existing gauges. Implementations are expected to be similar to counters.
+ */
+final class InMemoryMetrics implements Metrics {
+
+    /**
+     * Counters by name.
+     */
+    private final ConcurrentMap<String, InMemoryCounter> cnts = new ConcurrentHashMap<>();
+
+    @Override
+    public InMemoryCounter counter(final String name) {
+        return this.cnts.computeIfAbsent(name, ignored -> new InMemoryCounter());
+    }
+
+    @Override
+    public InMemoryGauge gauge(final String name) {
```

Comment:
@olegmoz l'd suggest returning `Gauge` and `Counter` instead of theirs implementation

Reply:
@olenagerasimova that's intended, if we take a counter from `InMemoryMetrics` then we can read it's value